Chris

Chrs987 opened the pull request:
Pushing the rough draft of my use-case, mis-use case, and prevention. Will work on diagram in a bit

Chrs987 committed:
Update Requirements for SSE.md

Updated title and link to test

Chrs987 committed:
Update Requirements for SSE.md

Updated information for Use Case, Misuse Case, and Prevention

Chrs987 committed:
Update Requirements for SSE.md

Adding information to Planning and Contribution

Chrs987 committed:
Update Requirements for SSE.md

Fixed spelling errors in Planning and Contribution

Chrs987 committed:
Add files via upload

Use-case2 diagram file

jmpinoy commented on Requirements for SSE.md:
```diff
@@ -4,15 +4,15 @@
 ### Case list
 
-#### TODO: Usecase 1 Link
+#### [Usecase 1](#use-case-3)
```
I made these changes to, hopefully github has a merge conflict resolver for us to pick which lines we want the master to have. Besides that everything else looks good.

Chrs987 replied:
Awesome sounds good! Im sure if worse case scenarios if they are just HREF links we can just edit them and clean it up tomorrow in the Master branch.

Chrs987 committed:
Update Requirements for SSE.md

Edited misuse-usecases, Observation of Security-Related Configuration and Installation Issues, and Planning and Contribution. Added Documentation Sources as well

Chrs987 committed:
Update Requirements for SSE.md

Changed image name from Use-Case2.png to Use-Case2.PNG because GitHub is case sensitive for how "PNG" is worded

Chrs987 committed:
Update Requirements for SSE.md

Changed image name from Use-Case2.png to Use-Case2.PNG because GitHub is case sensitive for how "PNG" is worded

jmpinoy commented on Requirements for SSE.md:
```diff
@@ -119,18 +118,45 @@ TODO: Insert Diagram photo
 ## Documentation Sources
 
-TODO: Insert documents referenced above (if any)
+* [IFTTT](https://ifttt.com/)
```
Can you add these links too, thanks.
[HA Add-on Communication] (https://developers.home-assistant.io/docs/add-ons/communication)
[HA Add-on Security] (https://developers.home-assistant.io/docs/add-ons/security/)
[HA Community Contribution Guidelines] (https://github.com/hassio-addons/repository/blob/master/CONTRIBUTING.md)

Chrs987 replied:
done

Chrs987 committed:
Update Requirements for SSE.md

added links per Josh comment

Chrs987 committed:
Update Requirements for SSE.md

fixed links

Jbibianochavez commented on Requirements for SSE.md:
```diff
@@ -37,22 +37,21 @@ TODO: Insert Diagram photo
+#### Updating Configuration.yaml Use Case
 
-TODO: Insert use-case info
+Bob was a user of a popular application called [IFTTT](https://ifttt.com/) and automated many of his IoT devices until they began their "Pro" subscription service which limited the number of integrations he could use unless he paid IFTTT's monthly subscription fee. Bob recently talked with his neighbor who shared a blog post about [HomeAssistant](https://blog.briancmoses.com/2020/09/ditching-iftt ...
```
Everything looks good. Reading through this I saw talking should be spoke. Also, no big deal but maybe break up the first sentence. maybe something like this. --> Bob was a user of a popular application called [IFTTT](https://ifttt.com/) and automated many of his IoT devices. They began offering their "Pro" subscription service which limited the number of integrations he could use unless he paid IFTTT's monthly subscription fee.

Chrs987 replied:
Fixed, nice catch thank you

Jbibianochavez commented on Requirements for SSE.md:
```diff
@@ -37,22 +37,21 @@ TODO: Insert Diagram photo
 #### Prevention/Security Requirement
 
-TODO: Assess alignment of security requirements derived from mis-use case analysis with advertised features of the open-source software. Review OSS project documentation and codebase to support your observations. Summarize your observations.
+To prevent the mis-use case, HomeAssisnt employs several safety measures. To prevent an attacker from gaining access to the Configurations.yaml file, the web interface is protected by a password and multi-factor authentication to prevent bruteforce attacks. In the event that a user will enable the Samba addon, Samba requires password authentication and the attacking machine to be on the same loca ...
```
Since Bob was giving out his main network password maybe he can set up a guest network to prevent this. Maybe this is out of scope. If so, please ignore.

Chrs987 replied:
Yeah, a guest network is not something offered by HA to prevent the misuse case. I also did not see anything about that in the documentation or any sort of "network segmentation" so I'm thinking it is out of scope for this assignment. Maybe something we could contribute to in their documentation at a later point.

Jbibianochavez commented on Requirements for SSE.md:
```diff
@@ -119,18 +118,47 @@ TODO: Insert Diagram photo
+
+When the user first goes to the Web UI, they are prompted to create the HA administrator account that has the ability to change everything. They will enter a 'Name' 'Username' and 'Password' that is used every time the user accesses HA. HA then asks for the devices primary location and states that this information is not shared outside of the users network. Next the user will begin onboarding (ad ...
+
+Lovelace offers users an [automation editor](https://www.home-assistant.io/getting-started/automation/) which gives users a clean and easy way to build their own automation. Here users can enable [Multi-factor authentication](https://www.home-assistant.io/docs/authentication/multi-factor-auth/) and ad an extra level of security to their account. The automation editor converts the users newly crea ...
```
Small typo: ad should be add

Jbibianochavez reviewed (commented):
Great work, a couple grammar fixes and it is good to go.

Chrs987 committed:
Update Requirements for SSE.md

Fixed typos per comments

Chrs987 committed:
Merge branch 'master' into Chris
